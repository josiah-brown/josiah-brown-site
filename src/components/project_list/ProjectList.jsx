import React from "react";
import ProjectSnippet from "../project_snippet/ProjectSnippet";
import { projects } from "../../data/siteContent";

const ProjectList = () => {
  return (
    <div
      data-testid="projects-list"
      className="flex flex-col space-y-4 lg:space-y-8 my-4 overflow-auto"
    >
      {projects.map((p) => {
        return <ProjectSnippet project={p} key={p.id} />;
      })}
    </div>
  );
};

export default ProjectList;
