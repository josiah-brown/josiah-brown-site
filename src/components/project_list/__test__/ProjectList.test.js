import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import ProjectList from "../ProjectList";

afterEach(cleanup);

test("renders project list without crash", () => {
  render(<ProjectList />);
});

test("renders project list correctly", () => {
  const { getByTestId } = render(<ProjectList />);
  expect(getByTestId("projects-list")).toBeVisible();
  expect(getByTestId("projects-list")).not.toBeEmptyDOMElement();
});

test("matches projects list snapshot 1", () => {
  const tree = renderer.create(<ProjectList />).toJSON();
  expect(tree).toMatchSnapshot();
});
