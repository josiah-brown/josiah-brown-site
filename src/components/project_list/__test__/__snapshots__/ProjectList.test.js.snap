// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches projects list snapshot 1 1`] = `
<div
  className="flex flex-col space-y-4 lg:space-y-8 my-4 overflow-auto"
  data-testid="projects-list"
>
  <div
    className="flex flex-col items-start border-black dark:border-white border-l-2 pl-2 group w-fit hover:cursor-pointer"
    data-testid="project-snippet-1"
    onPointerEnter={[Function]}
    onPointerLeave={[Function]}
  >
    <a
      data-testid="project-snippet-anchor"
      href="/projects/panther-city-leather"
    >
      <h3
        className="font-semibold text-xs md:text-sm lg:text-xl xl:text-2xl"
      >
        01.
      </h3>
      <div
        className="flex flex-row"
      >
        <h2
          className="italic font-light text-xl md:text-2xl lg:text-3xl xl:text-4xl"
        >
          Panther City Leather
        </h2>
        <svg
          className="lg:group-hover:-rotate-[135deg] lg:duration-200 my-auto text-2xl md:text-3xl lg:text-4xl xl:text-5xl"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="7"
            x2="17"
            y1="17"
            y2="7"
          />
          <polyline
            points="7 7 17 7 17 17"
          />
        </svg>
      </div>
      <p
        className="font-light text-xs md:text-sm lg:text-base xl:text-lg"
      >
        Full-Stack Web, Commerce.js, Stripe.js, Formik
      </p>
    </a>
    <div
      className="fixed invisible w-[200px] border-black dark:border-white border-2 pointer-events-none"
      id="img-1"
    >
      <img
        src="project-1.png"
      />
    </div>
  </div>
  <div
    className="flex flex-col items-start border-black dark:border-white border-l-2 pl-2 group w-fit hover:cursor-pointer"
    data-testid="project-snippet-2"
    onPointerEnter={[Function]}
    onPointerLeave={[Function]}
  >
    <a
      data-testid="project-snippet-anchor"
      href="/projects/portfolio"
    >
      <h3
        className="font-semibold text-xs md:text-sm lg:text-xl xl:text-2xl"
      >
        02.
      </h3>
      <div
        className="flex flex-row"
      >
        <h2
          className="italic font-light text-xl md:text-2xl lg:text-3xl xl:text-4xl"
        >
          Portfolio Site
        </h2>
        <svg
          className="lg:group-hover:-rotate-[135deg] lg:duration-200 my-auto text-2xl md:text-3xl lg:text-4xl xl:text-5xl"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="7"
            x2="17"
            y1="17"
            y2="7"
          />
          <polyline
            points="7 7 17 7 17 17"
          />
        </svg>
      </div>
      <p
        className="font-light text-xs md:text-sm lg:text-base xl:text-lg"
      >
        Full-Stack Web, Tailwind, Jest, P5.js, Figma
      </p>
    </a>
    <div
      className="fixed invisible w-[200px] border-black dark:border-white border-2 pointer-events-none"
      id="img-2"
    >
      <img
        src="project-2.png"
      />
    </div>
  </div>
  <div
    className="flex flex-col items-start border-black dark:border-white border-l-2 pl-2 group w-fit hover:cursor-pointer"
    data-testid="project-snippet-3"
    onPointerEnter={[Function]}
    onPointerLeave={[Function]}
  >
    <a
      data-testid="project-snippet-anchor"
      href="/projects/certificate-generator"
    >
      <h3
        className="font-semibold text-xs md:text-sm lg:text-xl xl:text-2xl"
      >
        03.
      </h3>
      <div
        className="flex flex-row"
      >
        <h2
          className="italic font-light text-xl md:text-2xl lg:text-3xl xl:text-4xl"
        >
          Diploma Generator App
        </h2>
        <svg
          className="lg:group-hover:-rotate-[135deg] lg:duration-200 my-auto text-2xl md:text-3xl lg:text-4xl xl:text-5xl"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="7"
            x2="17"
            y1="17"
            y2="7"
          />
          <polyline
            points="7 7 17 7 17 17"
          />
        </svg>
      </div>
      <p
        className="font-light text-xs md:text-sm lg:text-base xl:text-lg"
      >
        Desktop App, Python, PyPDF4
      </p>
    </a>
    <div
      className="fixed invisible w-[200px] border-black dark:border-white border-2 pointer-events-none"
      id="img-3"
    >
      <img
        src="project-3.png"
      />
    </div>
  </div>
  <div
    className="flex flex-col items-start border-black dark:border-white border-l-2 pl-2 group w-fit hover:cursor-pointer"
    data-testid="project-snippet-4"
    onPointerEnter={[Function]}
    onPointerLeave={[Function]}
  >
    <a
      data-testid="project-snippet-anchor"
      href="/projects/cipher-app"
    >
      <h3
        className="font-semibold text-xs md:text-sm lg:text-xl xl:text-2xl"
      >
        04.
      </h3>
      <div
        className="flex flex-row"
      >
        <h2
          className="italic font-light text-xl md:text-2xl lg:text-3xl xl:text-4xl"
        >
          Mobile Cipher App
        </h2>
        <svg
          className="lg:group-hover:-rotate-[135deg] lg:duration-200 my-auto text-2xl md:text-3xl lg:text-4xl xl:text-5xl"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="7"
            x2="17"
            y1="17"
            y2="7"
          />
          <polyline
            points="7 7 17 7 17 17"
          />
        </svg>
      </div>
      <p
        className="font-light text-xs md:text-sm lg:text-base xl:text-lg"
      >
        Mobile App, Python, Encryption
      </p>
    </a>
    <div
      className="fixed invisible w-[200px] border-black dark:border-white border-2 pointer-events-none"
      id="img-4"
    >
      <img
        src="project-4.jpg"
      />
    </div>
  </div>
</div>
`;
